src/en/authors-charm-actions.md

```diff
@@ -0,0 +1,233 @@
+# Actions for the Charm author
+
+---
+
+ - ### [Defining Actions](#defining-actions)
+    - #### [Requirements](#requirements)
+    - #### [Example Schema](#example-schema)
+ - ### [Params Transformation](#params-transformation)
+    - #### [Simple YAML transform example](#simple-yaml-transform-example)
+    - #### [More complex YAML transform example](#more-complex-yaml-transform-example)
+
+---
+
+## Defining Actions
+
+To define Actions ([see here for more about Actions](actions.html)) on a Charm, executables (scripts, binaries, etc.) for each 
+Action must be included in `/actions` in the Charm, and each named executable
+must be defined in a YAML map in `/actions.yaml` in the Charm.  [See below](#example-schema) for
+an example.
+
+### Requirements
+
+ - Each Action is defined as a top-level key of a YAML map, with the same name as
+the script it defines.
+
+ - The value of each Action must be a map, which should include a `description`
+key and may include a `params` key to define a schema.  If no `description` is
+given, a default empty description will be used.
+
+ - The value of `params`, if it is included, must be a YAML map.  Each key under
+`params` must be a string, and must have valid [JSON-Schema](http://json-schema.org/example2.html) as its value,
+transformed to a YAML map.  JSON-Schema may be nested to create complex
+schemas.
+
+ - JSON-Schema defines special keys such as `required` and `additionalProperties`,
+which may be given for the whole action at the same level as `description` and
+`params`, or within nested schemas at the usual level.
+
+ - At this time, the `$schema` key is not supported by Juju, as it may trigger
+resolution of remote objects.
+
+### Example Schema
+
+ > Charm dir
+```
+/actions/
```

Comment:
You'd want to show a tree output of a sample charm with all directory contents except actions stripped to keep with current formatting